runtime/common/src/paras_registrar.rs

```diff
@@ -1067,6 +1082,28 @@ mod tests {
 			// Data is swapped
 			assert_eq!(SwapData::get().get(&para_1).unwrap(), &1337);
 			assert_eq!(SwapData::get().get(&para_2).unwrap(), &69);
+
+			// Both paras initiate a swap
+            // Swap between parathread and parachain
+            assert_ok!(Registrar::swap(para_origin(para_1), para_1, para_2,));
+            assert_ok!(Registrar::swap(para_origin(para_2), para_2, para_1,));
+            assert_eq!(
+                System::events().last().unwrap(),
+                vec![EventRecord {
+                    phase: Phase::Initialization,
+                    event: Event::Registrar(paras_registrar::Event::Swapped {
+                        para_id: para_1,
+                        other_id: para_2
+                    }),
+                    topics: vec![],
+                }]
+                .last()
+                .unwrap()
+            );
```

Comment:
We did not have a test case for a swap between parathread and parachain which is one of the `else if` conditions here:
- https://github.com/paritytech/polkadot/blob/53cc5146bf505fbb2a110a815404703304df2c8e/runtime/common/src/paras_registrar.rs#L297-L300